src/mono/CMakeLists.txt

```diff
@@ -506,6 +506,14 @@ else()
   message(FATAL_ERROR "TARGET_ARCH='${TARGET_ARCH}' not supported.")
 endif()
 
+if(TARGET_ANDROID AND (TARGET_AMD64 OR TARGET_ARM64))
+  # Google requires all the native libraries to be aligned to 16 bytes (for 16k memory page size)
+  # This applies only to 64-bit binaries
+  if((TARGET_ARCH STREQUAL "arm64") OR (TARGET_ARCH STREQUAL "x86_64"))
+    add_link_options(LINKER:-z,max-page-size=16384)
+  endif()
+endif()
```

Comment:
in mono cmake TARGET is the cross-compiler target platform, and HOST is the platform where the runtime is running.  tl;dr you want `if (HOST_ANDROID AND (HOST_AMD64 OR HOST_ARM64))` etc